### ISSUE TITLE
publish_message throws Unable to identify task type because key was not found in the mesasge header: ''

### ISSUE THREAD

**Issue by nimphal** [chris104957/django-carrot#106]
I am using Django 2.1 and django-carrot 14.2 (latest available in PyPi). I have based my code on the instructions in the docs. 

```python
def my_task(**kwargs):
    return 'hello world'

publish_message(my_task, hello=True)
```

The task is correctly identified and pushed to Rabbit MQ, but I get the following error in monitor:

```
Traceback (most recent call last):
File "/usr/local/lib/python3.6/site-packages/carrot/consumer.py", line 300, in on_message
task_type = self.get_task_type(properties.headers, body)
File "/usr/local/lib/python3.6/site-packages/carrot/consumer.py", line 117, in get_task_type
return properties[self.serializer.type_header]
KeyError: ''
```
![carrot-monito-error](https://user-images.githubusercontent.com/4950775/54907908-97eb5780-4ede-11e9-89e1-23a594b9f5c3.png)

I have been digging in the code a bit, and it looks like there might be a discrepancy in how the type_header is called in the two message serializers. I have not validated this by pulling this repo and trying to debug, but I believe DefaultMessageSerializer is meant to override the BaseMessageSerializer https://github.com/chris104957/django-carrot/blob/35ec132ec07d4600035e2008c0d2db0e77335cc1/carrot/objects.py#L196 but for some reason that is not happening. It might be a discrepancy between calling variables with self and without in BaseMessageSerializer get_task:
```python
class BaseMessageSerializer(object):
    """
    A class that defines how to convert a RabbitMQ message into an executable python function from your Django project,
    and back again
    :param Message message: the RabbitMQ message
    """
    content_type = 'application/json'
    type_header, message_type = ('',) * 2
    task_get_attempts = 20

    def get_task(self, properties: pika.BasicProperties, body: bytes) -> Callable:
        """
        Identifies the python function to be executed from the content of the RabbitMQ message. By default, Carrot
        returns the value of the self.type_header header in the properties.
        Once this string has been found, carrot uses importlib to return a callable python function.
        """
        mod = '.'.join(properties.headers[self.type_header].split('.')[:-1])
        task = properties.headers[self.type_header].split('.')[-1]
        module = importlib.import_module(mod)
        func = getattr(module, task)
return func
```

This is mostly a guess and I have not validated it. I have not found anything listing this error in closed issues or in the documentation. It seems a lot of files got reset recently, so the history on github is also not very helpful. Any pointers would be greatly appreciated, I quite like the simplicity of this project and would love to use it!

**Comment by chris104957** [comment]
Can reproduce - will investigate

**Comment by chris104957** [comment]
So the problem was actually in `carrot.consumer`, this bit:
```python
class Consumer(threading.Thread):
    """... """
    serializer: Type[BaseMessageSerializer] = BaseMessageSerializer
```

The `BaseMessageSerializer` is not actually meant to be used like this, just extended. This issue got introduced when we added `mypy` and type checking. It's now been changed to this:
```python
class Consumer(threading.Thread):
    """... """
    serializer: Type[BaseMessageSerializer] = DefaultMessageSerializer
```

This fix has been made on a dev release, `1.4.3a1`. Can you install it (using the below command) and let me know if this fixes it?

```
pip install django-carrot==1.4.3a1 --no-cache
```

I'll remove the dev flags from this version number if it works

**Comment by nimphal** [comment]
That looks to be the thing, many thanks for looking into it so quickly!
